### PR TITLE
[ISSUE #3996] fix the acl content of TopicRequestHeader

### PR DESCRIPTION
**Pull request by chaiyx**
#3996

### CODE REVIEW

File: acl/src/main/java/org/apache/rocketmq/acl/common/AclClientRPCHook.java [hunk 1]
```diff
@@ -69,7 +72,11 @@ public void doAfterResponse(String remoteAddr, RemotingCommand request, Remoting
             if (null != header) {
                 Field[] fields = fieldCache.get(header.getClass());
                 if (null == fields) {
-                    fields = header.getClass().getDeclaredFields();
+                    Set<Field> fieldList = new HashSet<Field>();
```
Comment (caigy) [inline acl/src/main/java/org/apache/rocketmq/acl/common/AclClientRPCHook.java:75]: `parseRequestContent` is used for signature checks, so the order of fields matters. I think a sorted set should be adopted here.

**Review by caigy — CHANGES_REQUESTED** [review]

File: acl/src/main/java/org/apache/rocketmq/acl/common/AclClientRPCHook.java (hunk 1 above)
Comment (chaiyx) [reply acl/src/main/java/org/apache/rocketmq/acl/common/AclClientRPCHook.java:75]: at the end of the function, the value of the header will be put into a SortedMap. so it doesn't matter to use HashSet here.

Comment (caigy) [reply acl/src/main/java/org/apache/rocketmq/acl/common/AclClientRPCHook.java:75]: But the order of fields put into returning SortedMap still depends on the order of `fieldList`.

**Comment by RongtongJin** [comment]
Hi @chaiyx  Could you resubmit your PR to the 5.0.0-beta branch because 5.0.0-alpha already release?

**Pull request closed without merging**